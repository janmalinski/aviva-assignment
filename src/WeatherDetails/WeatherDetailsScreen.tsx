import React, { useCallback } from 'react';
import { StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import WeatherDetails from '../components/WeatherDetails';
import { NavigationProp, RouteProp, useNavigation, useRoute } from '@react-navigation/native';
import { TRootNavigatorParams } from '../navigation/RootNavigator';
import Button from '../components/Button';

const WeatherDetailsScreen = () => {
  const route = useRoute<RouteProp<TRootNavigatorParams, "WeatherDetails">>();
  const navigation = useNavigation<NavigationProp<TRootNavigatorParams, "WeatherDetails">>();

  const navigateToHome = useCallback(() => {
    navigation.navigate("Home");
  }, [navigation]);

  return (
    <SafeAreaView style={styles.container} edges={['top']}>
      <WeatherDetails screen='WeatherDetails' weatherData={{...route.params}}  />
      <Button title='Back' onPress={navigateToHome}/>
    </SafeAreaView>
  )
}

export default WeatherDetailsScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 40,
    backgroundColor: '#e4f6f4'
  },
})