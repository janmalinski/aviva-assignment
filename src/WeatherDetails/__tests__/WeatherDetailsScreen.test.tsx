import React from "react";
import { render } from "@testing-library/react-native";
import WeatherDetailsScreen from "../WeatherDetailsScreen";

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: jest.fn(),
      }),
      useRoute: () => ({
        params : {
        city: 'PoznaÅ„',
        humidity: 54,
        pressure: 1000,
        temp: 19,
        feels_like: 18,
        temp_max: 20,
        temp_min: 18,
        description: 'Light inisity showers',
        icon: '10d',
        }
      })
    };
  });


describe('WeatherDetailsScreen', () => {
    it('Should render correctly', () => {
        render(<WeatherDetailsScreen />)
    });

    it('Should match a snapshot', () => {
        const elJson = render(<WeatherDetailsScreen />).toJSON();
        expect(elJson).toMatchSnapshot(); 
    });

});



