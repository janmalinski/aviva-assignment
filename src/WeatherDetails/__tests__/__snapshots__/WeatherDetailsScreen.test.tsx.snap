// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetailsScreen Should match a snapshot 1`] = `
<RNCSafeAreaView
  edges={
    [
      "top",
    ]
  }
  style={
    {
      "backgroundColor": "#e4f6f4",
      "flex": 1,
      "paddingHorizontal": 40,
    }
  }
>
  <View
    style={
      {
        "marginTop": 0,
      }
    }
    testID="weather-details"
  >
    <View
      style={
        {
          "marginBottom": 40,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
              "gap": 10,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "fontFamily": "Actor-Regular",
              "fontSize": 22,
            }
          }
        >
          Poznań
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            {
              "uri": "http://openweathermap.org/img/wn/10d@2x.png",
            }
          }
          style={
            [
              {
                "height": 100,
                "marginTop": -15,
                "width": 100,
              },
              {
                "marginLeft": -20,
              },
            ]
          }
        />
        <Text
          style={
            [
              {
                "fontFamily": "Actor-Regular",
                "fontSize": 22,
              },
              {
                "marginLeft": 5,
                "marginTop": 5,
              },
            ]
          }
        >
          19
          ˚
        </Text>
        <View
          style={
            {
              "backgroundColor": "#000",
              "height": "80%",
              "marginHorizontal": 10,
              "width": 1,
            }
          }
        />
        <Text
          style={
            [
              {
                "fontFamily": "Actor-Regular",
                "fontSize": 18,
                "lineHeight": 24,
              },
              {
                "flex": 1,
                "flexWrap": "wrap",
                "marginTop": 5,
              },
            ]
          }
        >
          Light inisity showers
          .
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Actor-Regular",
              "fontSize": 18,
              "lineHeight": 24,
            }
          }
        >
          Maximum temprature: 
          20
          ˚
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Actor-Regular",
              "fontSize": 18,
              "lineHeight": 24,
            }
          }
        >
          Minimum temprature: 
          18
          ˚
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Actor-Regular",
              "fontSize": 18,
              "lineHeight": 24,
            }
          }
        >
          Feels like: 
          18
          ˚
        </Text>
      </View>
      <Text
        style={
          {
            "fontFamily": "Actor-Regular",
            "fontSize": 18,
            "lineHeight": 24,
          }
        }
      >
        Humidity: 
        54
        %
      </Text>
      <Text
        style={
          {
            "fontFamily": "Actor-Regular",
            "fontSize": 18,
            "lineHeight": 24,
          }
        }
      >
        Pressure: 
        1000
        mb
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 4,
        "borderWidth": 1,
        "justifyContent": "center",
        "minHeight": 56,
        "opacity": 1,
        "paddingHorizontal": 15,
      }
    }
  >
    <Text>
      Back
    </Text>
  </View>
</RNCSafeAreaView>
`;
